src/ert/scheduler/scheduler.py

```diff
@@ -214,35 +214,36 @@ async def gather_realization_jobs() -> list[BaseException | None]:
                 for scheduling_task in scheduling_tasks:
                     scheduling_task.cancel()
 
-        job_results: Optional[list[BaseException | None]] = None
         try:
             # there are two types of tasks and each type is handled differently:
             # -`gather_realization_jobs` does not raise; rather, each job task's result
             # is collected into the returning list. Exceptions from the job tasks are
             # handled in the `else` branch after the evaluation has stopped.
             # -If exception occurs, it must necessarily come from `scheduling tasks`
-            job_results = (
-                await asyncio.gather(gather_realization_jobs(), *scheduling_tasks)
-            )[0]
+
+            await asyncio.gather(gather_realization_jobs(), *scheduling_tasks)
```

Comment:
This will raise an asyncio.CancelledError because of line 215. We can instead find all the job_results by looking directly at the tasks after gather-gather has finished.